scripts/utils/scratch.ts

```diff
@@ -0,0 +1,259 @@
+    epochsPerFrame: number;
+  };
+  validatorsExitBusOracle: {
+    consensusVersion: number;
+    maxValidatorsPerRequest: number;
+    maxExitRequestsLimit: number;
+    exitsPerFrame: number;
+    frameDurationInSec: number;
+  };
+  depositSecurityModule: {
+    maxOperatorsPerUnvetting: number;
+    pauseIntentValidityPeriodBlocks: number;
+    usePredefinedAddressInstead?: string;
+  };
+  oracleReportSanityChecker: {
+    exitedValidatorsPerDayLimit: number;
+    appearedValidatorsPerDayLimit: number;
+    deprecatedOneOffCLBalanceDecreaseBPLimit: number;
+    annualBalanceIncreaseBPLimit: number;
+    simulatedShareRateDeviationBPLimit: number;
+    maxValidatorExitRequestsPerReport: number;
+    maxItemsPerExtraDataTransaction: number;
+    maxNodeOperatorsPerExtraDataItem: number;
+    requestTimestampMargin: number;
+    maxPositiveTokenRebase: number;
+    initialSlashingAmountPWei: number;
+    inactivityPenaltiesAmountPWei: number;
+    clBalanceOraclesErrorUpperBPLimit: number;
+  };
+  oracleDaemonConfig: {
+    NORMALIZED_CL_REWARD_PER_EPOCH: number;
+    NORMALIZED_CL_REWARD_MISTAKE_RATE_BP: number;
+    REBASE_CHECK_NEAREST_EPOCH_DISTANCE: number;
+    REBASE_CHECK_DISTANT_EPOCH_DISTANCE: number;
+    VALIDATOR_DELAYED_TIMEOUT_IN_SLOTS: number;
+    VALIDATOR_DELINQUENT_TIMEOUT_IN_SLOTS: number;
+    NODE_OPERATOR_NETWORK_PENETRATION_THRESHOLD_BP: number;
+    PREDICTION_DURATION_IN_SLOTS: number;
+    FINALIZATION_MAX_NEGATIVE_REBASE_EPOCH_SHIFT: number;
+  };
+  nodeOperatorsRegistry: {
+    stakingModuleName: string;
+    stakingModuleTypeId: string;
+    stuckPenaltyDelay: number;
+  };
+  simpleDvt: {
+    stakingModuleName: string;
+    stakingModuleTypeId: string;
+    stuckPenaltyDelay: number;
+  };
+  withdrawalQueueERC721: {
+    name: string;
+    symbol: string;
+  };
+  validatorExitDelayVerifier: {
+    gIFirstValidatorPrev: string;
+    gIFirstValidatorCurr: string;
+    gIFirstHistoricalSummaryPrev: string;
+    gIFirstHistoricalSummaryCurr: string;
+    gIFirstBlockRootInSummaryPrev: string;
+    gIFirstBlockRootInSummaryCurr: string;
+  };
+  triggerableWithdrawalsGateway: {
+    maxExitRequestsLimit: number;
+    exitsPerFrame: number;
+    frameDurationInSec: number;
+  };
+  predepositGuarantee: {
+    gIndex: string;
+    gIndexAfterChange: string;
+    changeSlot: number;
+  };
+  operatorGrid: {
+    defaultTierParams: {
+      shareLimitInEther: string;
+      reserveRatioBP: number;
+      forcedRebalanceThresholdBP: number;
+      infraFeeBP: number;
+      liquidityFeeBP: number;
+      reservationFeeBP: number;
+    };
+  };
+}
+
+export function readScratchParameters(): ScratchParameters {
+  const rawData = fs.readFileSync(SCRATCH_DEPLOY_CONFIG, "utf8");
+  return toml.parse(rawData) as unknown as ScratchParameters;
+}
+
+// Convert TOML scratch parameters to deployment state format
+export function scratchParametersToDeploymentState(params: ScratchParameters): Record<string, unknown> {
```

Comment:
Will it break if something is missing in the file? Maybe use [Zod](https://zod.dev/) to verify that the object is full?